crates/mitex/src/lib.rs

```diff
@@ -424,22 +553,23 @@ impl MathConverter {
     }
 }
 
-struct TypstMathRepr(LatexSyntaxElem, CommandSpec, Rc<RefCell<String>>);
+struct TypstRepr(LatexSyntaxElem, LaTeXMode, CommandSpec, Rc<RefCell<String>>);
 
-impl fmt::Display for TypstMathRepr {
+impl fmt::Display for TypstRepr {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let mut ctx = MathConverter::new();
-        if let Err(e) = ctx.convert(f, self.0.clone(), &self.1) {
-            self.2.borrow_mut().push_str(&e.to_string());
+        let mut ctx = Converter::new(self.1);
+        if let Err(e) = ctx.convert(f, self.0.clone(), &self.2) {
+            self.3.borrow_mut().push_str(&e.to_string());
```

Comment:
There are two many fields in a single tuple. I would prefer to refactoring `TypstMathRepr` as 
```rs
struct TypstRepr  {
  elem: LatexSyntaxElem,
  mode: LaTeXMode,
  spec: CommandSpec,
  error: Rc<RefCell<String>>
}
```